api/oc_push.c

```diff
@@ -0,0 +1,2628 @@
+        break;
+      }
+
+      /*
+       * client에서 POST 하는 루틴 참조할 것 (client_multithread_linux.c 참고)
+       */
+      /*
+       * 1. find `notification selector` which monitors `src_rsc` from
+       * `ns_col_list`
+       * 2. post UPDATE by using URI, endpoint (use oc_sting_to_endpoint())
+       */
+      if (oc_init_post(oc_string(ns_instance->targetpath),
+                       &ns_instance->pushtarget_ep, "if=oic.if.rw",
+                       &response_to_push_rsc, LOW_QOS, NULL)) {
+        /*
+         * add other properties than "rep" object of "oic.r.pushpayload"
+         * Resource here. payload_builder() only adds "rep" object.
+         *
+         * payload_builder() doesn't need to have "oc_rep_start_root_object()"
+         * and "oc_rep_end_root_object()" they should be added here...
+         */
+
+        oc_rep_begin_root_object();
+
+        /* anchor */
+        oc_uuid_to_str(oc_core_get_device_id(ns_instance->resource->device), di,
+                       OC_UUID_LEN);
+        oc_rep_set_text_string(root, anchor, di);
+
+        /* href (option) */
+        if (oc_string(ns_instance->phref) &&
+            strcmp(oc_string(ns_instance->phref), "")) {
+          oc_rep_set_text_string(root, href, oc_string(ns_instance->phref));
+        }
+
+        /* rt (array) */
+        oc_rep_open_array(root, rt);
+        for (size_t i = 0;
+             i < oc_string_array_get_allocated_size(src_rsc->types); i++) {
+          oc_rep_add_text_string(rt,
+                                 oc_string_array_get_item(src_rsc->types, i));
+        }
+        oc_rep_close_array(root, rt);
+
+        /* if (array) */
+        oc_core_encode_interfaces_mask(oc_rep_object(root),
+                                       src_rsc->interfaces);
+
+        src_rsc->payload_builder();
+
+        oc_rep_end_root_object();
+
+        if (oc_do_post())
+          OC_PUSH_DBG("Sent POST request");
+        else
+          OC_PUSH_ERR("Could not send POST");
+      } else {
+        OC_PUSH_ERR("Could not init POST");
+      }
+    }
+  }
+
+  OC_PROCESS_END();
+}
+
+/*
+ * @brief check if any of source array is part of target array
+ * @param target
+ * @param source
+ * @return
+ * 			0: any of source is not part of target
+ * 			1: any of source is part of target
+ */
+static bool
+_check_string_array_inclusion(oc_string_array_t *target,
+                              oc_string_array_t *source)
+{
+  int i, j;
+  int src_len, tgt_len;
+  bool result = 0;
+
+  tgt_len = oc_string_array_get_allocated_size(*target);
+  src_len = oc_string_array_get_allocated_size(*source);
+
+  if (!tgt_len || !src_len) {
+    OC_PUSH_DBG("source or target string array is empty!");
+    return result;
+  }
+
+  for (i = 0; i < src_len; i++) {
+    for (j = 0; j < tgt_len; j++) {
+      if (!strcmp(oc_string_array_get_item(*source, i),
+                  oc_string_array_get_item(*target, j))) {
+        result = 1;
+        break;
```

Comment:
You can just:

```
return true;
```

here and then you don't need the two breaks.

Reply:
agree, fixed